src/libprojectM/Renderer/MilkdropNoise.cpp

```diff
@@ -1,42 +1,131 @@
 #include "MilkdropNoise.hpp"
 
+#include "projectM-opengl.h"
+
 #include <chrono>
 #include <random>
 
+// Missing in macOS SDK. Query will most certainly fail, but then use the default format.
+#ifndef GL_TEXTURE_IMAGE_FORMAT
+#define GL_TEXTURE_IMAGE_FORMAT 0x828F
+#endif
+
 namespace libprojectM {
 namespace Renderer {
 
-MilkdropNoise::MilkdropNoise()
+auto MilkdropNoise::LowQuality() -> std::shared_ptr<Texture>
+{
+    GLuint texture{};
+    {
```

Comment:
Any particular reason this is in a separate block?

Reply:
Slightly reduces the lifetime of the `textureData` vector, which is only required until after the `glTexImageXD` call. Not really a concern though as the method exits shortly after, if you find it less readable this way I can just remove the blocks.

Reply:
Not really a big deal either way, just making sure it was intentional. It's consistent with other places in the code so I say leave it.